### PR TITLE
Reduces log clutter

### LINKED ISSUE #907: Consensus, 253 logs

### PR DESCRIPTION
**Pull request by lasarojc**
Closes #907
This PR downgrades some `Info` logs that are relevant to developers but shouldn't matter to validators to `Debug` logs.
Most of these logs are on the `consensus` package.

On the positive side, there is less information overhead to operators and less storage is needed for saving logs during normal operation.
On the negative side, if debugging is needed, the log level must be changed after the need is detected, which may be too late to capture rare occurrences of bugs.

During your reviews, please let me know if any of the demoted logs have been useful in previous debug sessions and if you think they should not be demoted. 

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: consensus/reactor.go [hunk 1]
```diff
@@ -72,7 +72,8 @@ func NewReactor(consensusState *State, waitSync bool, options ...ReactorOption)
 // OnStart implements BaseService by subscribing to events, which later will be
 // broadcasted to other peers and starting state if we're not in block sync.
 func (conR *Reactor) OnStart() error {
-	conR.Logger.Info("Reactor ", "waitSync", conR.WaitSync())
+	conR.Logger.Debug("Consensus OnStart")
+	conR.Logger.Debug("waitSync", conR.WaitSync())
```
Comment (thanethomson) [inline consensus/reactor.go:76]: Seems like it should still be a single log line, should it not?

File: consensus/reactor.go [hunk 2]
```diff
@@ -105,7 +107,7 @@ func (conR *Reactor) OnStop() {
 // SwitchToConsensus switches from block_sync mode to consensus mode.
 // It resets the state, turns off block_sync, and starts the consensus state-machine
 func (conR *Reactor) SwitchToConsensus(state sm.State, skipWAL bool) {
-	conR.Logger.Info("SwitchToConsensus")
+	conR.Logger.Debug("Consensus SwitchToConsensus")
```
Comment (thanethomson) [inline consensus/reactor.go:110]: This seems like it should still be at info level to assist with debugging (assuming that most operators will always have their logs at info level and above, and never at debug level).

**Review by thanethomson — COMMENTED** [review]
While I agree with many of these changes (especially the ones that've been promoted from info to error) I'm not 100% sure whether operators rely on some of the logs that're being dropped to debug level. Also, operators have asked us for _more_ information at info level from consensus (and much less from P2P). It may be a good idea to chat with some of them to get an idea of what they want from consensus (perhaps in a follow-up PR?).

File: consensus/replay.go [hunk 3]
```diff
@@ -489,6 +488,10 @@ func (h *Handshaker) replayBlocks(
 			return nil, err
 		}
 
+		if i == firstBlock || i%1000 == 0 || i == finalBlock {
+			h.logger.Info("executed blocks up to", "height", block.Height, "app_hash", appHash)
```
[the remark this answers is not in the document]
Comment (mzabaluev) [reply consensus/replay.go:492]: In Rust frameworks like tracing and in general (e.g. in journald), the convention is not to have the message capitalized. In commonly used log printout formats, the message is prefixed with ancillary information (timestamp, severity, etc.), so the capitalization will stick out.

File: consensus/reactor.go (hunk 2 above)
Comment (cason) [reply consensus/reactor.go:110]: Agreed. This message should be printed only once, which for me is a clear case for Info level.

File: consensus/reactor.go [hunk 4]
```diff
@@ -650,12 +654,12 @@ func (conR *Reactor) gossipDataForCatchup(logger log.Logger, rs *cstypes.RoundSt
...
 		blockMeta := conR.conS.blockStore.LoadBlockMeta(prs.Height)
 		if blockMeta == nil {
-			logger.Error("Failed to load block meta", "ourHeight", rs.Height,
+			logger.Error("Failed to load block meta", "rs.height", rs.Height,
 				"blockstoreBase", conR.conS.blockStore.Base(), "blockstoreHeight", conR.conS.blockStore.Height())
 			time.Sleep(conR.conS.config.PeerGossipSleepDuration)
 			return
 		} else if !blockMeta.BlockID.PartSetHeader.Equals(prs.ProposalBlockPartSetHeader) {
-			logger.Info("Peer ProposalBlockPartSetHeader mismatch, sleeping",
+			logger.Error("Peer ProposalBlockPartSetHeader mismatch, sleeping",
```
Comment (cason) [inline consensus/reactor.go:662]: I would not put this an error. By the way we set those fields referring to a peer (they are hints), we cannot say that the peer is misbehaving when sending us the wrong block.

File: consensus/replay.go [hunk 5]
```diff
@@ -137,7 +137,7 @@ func (cs *State) catchupReplay(csHeight int64) error {
 	}
 	defer gr.Close()
 
-	cs.Logger.Info("Catchup by replaying consensus messages", "height", csHeight)
+	cs.Logger.Debug("Catchup by replaying consensus messages", "height", csHeight)
```
Comment (cason) [inline consensus/replay.go:140]: This again is a message that should be printed only once, Info level looks more appropriate.

File: consensus/state.go [hunk 6]
```diff
@@ -1603,7 +1603,7 @@ func (cs *State) enterCommit(height int64, commitRound int32) {
 	// If we don't have the block being committed, set up to get it.
 	if !cs.ProposalBlock.HashesTo(blockID.Hash) {
 		if !cs.ProposalBlockParts.HasHeader(blockID.PartSetHeader) {
-			logger.Info(
+			logger.Debug(
```
Comment (cason) [inline consensus/state.go:1606]: This looks to me an unlikely situation. I would preserve info, even warning if we have that level.

File: consensus/ticker.go [hunk 7]
```diff
@@ -121,7 +121,7 @@ func (t *timeoutTicker) timeoutRoutine() {
 			t.timer.Reset(ti.Duration)
 			t.Logger.Debug("Scheduled timeout", "dur", ti.Duration, "height", ti.Height, "round", ti.Round, "step", ti.Step)
 		case <-t.timer.C:
-			t.Logger.Info("Timed out", "dur", ti.Duration, "height", ti.Height, "round", ti.Round, "step", ti.Step)
+			t.Logger.Debug("Timed out", "dur", ti.Duration, "height", ti.Height, "round", ti.Round, "step", ti.Step)
```
Comment (cason) [inline consensus/ticker.go:124]: We need to make timeouts logged when they are processed, not when they are produced. So while I agree with this change of level, we need to turn into Info when a timeout produce an operation state change.

File: state/execution.go [hunk 8]
```diff
@@ -696,7 +696,7 @@ func ExecCommitBlock(
 		return nil, fmt.Errorf("expected tx results length to match size of transactions in block. Expected %d, got %d", len(block.Data.Txs), len(resp.TxResults))
 	}
 
-	logger.Info("executed block", "height", block.Height, "app_hash", resp.AppHash)
+	logger.Debug("executed block", "height", block.Height, "app_hash", resp.AppHash)
```
[the remark this answers is not in the document]
Comment (cason) [reply state/execution.go:699]: Agreed.

**Review by cason — COMMENTED** [review]
While this is a relevant issue to address, I wonder whether we need some more information in order to decide which logs and levels are relevant.

For instance, we should have a document displaying all log messages produced in Info and Debug level for a successful ("good") height, then when something odd happens (there are several situations to cover). With this overall output or representation we could discuss which logs are redundant, unclear, and which information is indeed missing.

Finally, I would rename the PR to make clear it refers to the consensus module/reactor.

File: consensus/replay.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (adizere) [reply consensus/replay.go:492]: > the convention is not to have the message capitalized. In commonly used log printout formats, the message is prefixed with ancillary information (timestamp, severity, etc.), so the capitalization will stick out.

+1

**Comment by lasarojc** [comment]
Relates to tendermint/tendermint#6987

File: consensus/state.go [hunk 9]
```diff
@@ -1899,7 +1899,7 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal) error {
 		cs.ProposalBlockParts = types.NewPartSetFromHeader(proposal.BlockID.PartSetHeader)
 	}
 
-	cs.Logger.Info("received proposal", "proposal", proposal)
+	cs.Logger.Debug("received proposal", "proposal", proposal)
```
Comment (sergio-mena) [inline consensus/state.go:1902]: This turned out to be crucial to day, so please leave it in Info

File: state/execution.go [hunk 10]
```diff
@@ -220,7 +220,7 @@ func (blockExec *BlockExecutor) ApplyBlock(
 		return state, err
 	}
 
-	blockExec.logger.Info(
+	blockExec.logger.Debug(
```
Comment (sergio-mena) [inline state/execution.go:223]: This one also was useful today (well, actually its `v0.34.x` counterpart: "executed block") to delimit the logs that were coming from the block execution itself. I think it should stay in Info

File: state/execution.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (sergio-mena) [reply state/execution.go:699]: I would change the message to be similar to "finalized block" above (it is its "handshake" counterpart), something like:
```
	blockExec.logger.Info(
		"finalized block (replay blocks)",
		"height", block.Height,
		"num_txs_res", len(resp.TxResults),
		"num_val_updates", len(resp.ValidatorUpdates),
		"block_app_hash", fmt.Sprintf("%X", resp.AppHash),
	)
```

File: state/txindex/indexer_service.go [hunk 11]
```diff
@@ -99,7 +99,7 @@ func (is *IndexerService) OnStart() error {
 						return
 					}
 				} else {
-					is.Logger.Info("indexed block events", "height", height)
+					is.Logger.Debug("indexed block events", "height", height)
```
Comment (sergio-mena) [inline state/txindex/indexer_service.go:102]: This one I saw it today quite often in the logs but I didn't really find it useful. I'd agree on "demoting" it to Debug

**Review by sergio-mena — COMMENTED** [review]
We've had a troubleshooting session today, trying to find out why, and under which conditions, there were frequent heights decided in round 1.
I've been exposed to the consensus logs for that troubleshooting session.
Since I have the logs I found (and didn't find) useful fresh in my mind, I decided to come back to this PR and provide first-hand feedback.

**Comment by lasarojc** [comment]
Thanks, @sergio-mena 
I should resume working on this PR soon.

**Comment by melekes** [comment]
@lasarojc any plans to resume this?

**Comment by melekes** [comment]
Let's revisit this PR when we have them then.

**Pull request closed without merging**